### PR TITLE
DDP-5058 squid: add fluentd logging config

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Add logging configuration for Squid and scripts to setup logging agent on proxy VM.

To deploy this to an existing proxy VM:
1. Run `install-logging-agent.sh` script.
2. Run `deploy-logging-config.sh` script.
3. Open Stackdriver logging dashboard and verify logs are showing up.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
    - Need to run the scripts.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [x] Releasing these changes requires special handling and I have documented the special procedures in the release plan document
    - Need to run the scripts.

### CODE REVIEW

File: pepper-apis/gae-egress-proxy/fluentd-logging.conf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+</source>
+
+<source>
+  @type tail
+  <parse>
+    @type regexp
+
+    # Parse entire line as `message` but also pull out individual fields.
+    # All of these will end up in the `jsonPayload` in Stackdriver log entry.
+    expression /^(?<message>(?<time>[0-9]+[\.0-9]*) +(?<elapsed>[0-9]+) (?<source_ip>[0-9\.]+) (?<action>[A-Z_]+)\/(?<status_code>[0-9]+) (?<size>[0-9]+) (?<method>[A-Z]+) (?<url>[^ ]+) (?<username>[^ ]+) (?<peer_status>[^ ]+)/(?<peer_ip>[^ ]+) (?<mime>[^ ]+))/
```
Comment (yufengwng) [inline pepper-apis/gae-egress-proxy/fluentd-logging.conf:28]: This is the default log format, per the documentation linked to in the ticket. Here's a nice little tool to try out the regex matching: http://fluentular.herokuapp.com/.

Comment (zyme) [reply pepper-apis/gae-egress-proxy/fluentd-logging.conf:28]: Very cool!  Is there a way to [encode fields](https://cloud.google.com/error-reporting/docs/formatting-error-messages) to trigger error reporting?

Comment (yufengwng) [reply pepper-apis/gae-egress-proxy/fluentd-logging.conf:28]: Just so I understand: is the idea to add additional fields for Error Reporting based on some conditional logic (e.g. if status_code is 500)? I think this should be possible using fluentd's filters, but I'll need to consult the documentation.

However, I'm not too sure about this endeavor at this moment. I'm not sure what in the logs will constitute something worthy of Error Reporting. For example, we can go off of the status_code, but if status_code is 5xx then shouldn't the clients report the errors and so we'll pick up on that either from `pepper-backend` or `housekeeping`? Certainly, if we find something interesting in the logs, we can add integration for it!

**Comment by yufengwng** [comment]
I think I'll just get this into the upcoming release instead of a hotfix. I don't think it's that urgent.

File: pepper-apis/gae-egress-proxy/fluentd-logging.conf (hunk 1 above)
Comment (zyme) [reply pepper-apis/gae-egress-proxy/fluentd-logging.conf:28]: Sounds like something for later, after we get some road time with the logs more generally.  All good!

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**